src/algebra/gcd_monoid.lean

```diff
@@ -592,14 +592,14 @@ begin
   cases hi.is_unit_or_is_unit hy with hu hu; cases hu with u hu,
   { right, transitivity (gcd (x * b) (a * b)), apply dvd_gcd (dvd_mul_right x b) h,
     rw gcd_mul_right, rw ← hu,
-    apply dvd_of_associated, transitivity (normalize b), symmetry, use u, apply mul_comm,
+    apply associated.dvd, transitivity (normalize b), symmetry, use u, apply mul_comm,
     apply normalize_associated, },
   { left, rw [hy, ← hu],
-    transitivity, {apply dvd_of_associated, symmetry, use u}, apply gcd_dvd_right, }
+    transitivity, { apply associated.dvd, symmetry, use u }, apply gcd_dvd_right, }
 end ⟩⟩
 
 theorem irreducible_iff_prime {p : α} : irreducible p ↔ prime p :=
-⟨prime_of_irreducible, irreducible_of_prime⟩
+⟨prime_of_irreducible, prime.irreducible⟩
```

Comment:
Definitely change this one too!
```suggestion
⟨irreducible.prime, prime.irreducible⟩
```
(l.588 is out of the scope of the diff, so I can't suggest on the statement `prime_of_irreducible` directly)

Reply:
That won't work with dot notation, since it's declared in the `gcd_monoid` namespace. (I don't think `open gcd_monoid` would help either.)

Reply:
Arf, yeah :/
But does it really need to be under the `gcd_monoid` namespace? Are there other ways of getting primality out of irreducibility?

Reply:
I don't know from the top of my head any interesting case other than `gcd_monoid` where all irreducibles are prime, but I don't see `irreducible → prime` implying the existence of a gcd either.

Reply:
Ah, I found the condition: being a [Schreier domain](https://en.wikipedia.org/wiki/Schreier_domain)

Reply:
Ahah, great find! Then if we ever define Schreier domains in mathlib, this one could be generalised. Could you then move it out of the `gcd_monoid` namespace and put a TODO for the generalisation?

Reply:
Sure, I guess I don't have strong feelings on whether to keep this namespaced until we define Schreier domains (monoids?).